lib/smartcar_error.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+# Custom SmartcarError class to represent errors from Smartcar APIs.
+class SmartcarError < StandardError
+  attr_reader :error, :code, :status_code, :request_id, :type, :description, :doc_url, :resolution, :detail
```

Comment:
I think there's a small thing that got missed here, but could be up for debate.

For v1.0 errors and for auth errors  should we set the `error` property or should we set the `type` property? The mock factory I wrote in the spec coerced `error` into `type`